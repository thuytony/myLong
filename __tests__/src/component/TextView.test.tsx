import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import { TextView } from '@components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const instance = renderer.create(<TextView />);
  expect(instance.toJSON()).toMatchSnapshot();
});

it('renders with text', () => {
  const instance = renderer.create(<TextView>test</TextView>);
  const textView = instance.root.findByType(Text);
  expect(textView.props.children).toBe('test');
});

it('test style h6', () => {
  const instance = renderer.create(<TextView h6>test</TextView>);
  const textView = instance.root.findByType(Text);
  expect(textView.props.style).toHaveProperty('fontFamily', 'Avenir-Medium');
  expect(textView.props.style).toHaveProperty('fontSize', 16);
  expect(textView.props.style).toHaveProperty('lineHeight', 20);
  expect(textView.props.style).toHaveProperty('fontWeight', '500');
  expect(textView.props.style).toHaveProperty('color', '#03606A');
});