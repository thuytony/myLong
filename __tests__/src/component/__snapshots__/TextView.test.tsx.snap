// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#03606A",
      "fontFamily": "Avenir-Medium",
      "fontSize": 14,
    }
  }
/>
`;
